configs/example/gem5_library/fdp-hello.py

```diff
@@ -0,0 +1,333 @@
+    cpu_type=CPUTypes.O3, isa=isa_choices[args.isa], num_cores=1
+)
+cpu = processor.cores[0].core
+
+
+# We need to configure the decoupled front-end with some specific parameters.
+# First the fetch buffer and fetch target size. We want double the size of
+# the fetch buffer to be able to run ahead of fetch
+cpu.fetchBufferSize = 16
+cpu.fetchTargetWidth = 32
+
+# The decoupled front-end leverages the BTB to find branches in the fetch
+# stream. Starting from the end of the last fetch target it will search
+# all addresses until a hit. However, for fixed size instruction architectures
+# like ARM only every n-th address must be checked.
+if args.isa == "Arm":
+    cpu.minInstSize = (
+        4  # Note Arm has a 2 byte thumb mode but we ignore it here
+    )
+elif args.isa == "RiscV":
+    cpu.minInstSize = 4  # RiscV has a 2 byte compressed mode.
+else:  # Variable length ISA (x86) must search every byte
+    cpu.minInstSize = 1
+
+# The `decoupledFrontEnd` parameter enables the decoupled front-end.
+# Disable it to get the baseline.
+if args.disable_fdp:
+    cpu.decoupledFrontEnd = False
+else:
+    cpu.decoupledFrontEnd = True
+
+
+## Branch Predictor Unit --------------------------------
+# The decoupled front-end relies on the BTB to discover
+# branches and uses `takenOnlyHistory`.
+# Also at the moment only LTAGE and TAGE-SC-L are supported.
+class BTB(SimpleBTB):
+    numEntries = 16 * 1024
+    associativity = 8
+
+
+class BPU(LTAGE):
+    instShiftAmt = 2 if args.isa == "Arm" else 0
+    btb = BTB()
+    requiresBTBHit = True
+    takenOnlyHistory = True
+
+
+# Configure the branch predictor
+cpu.branchPred = BPU()
+
+
+print(
+    "Running {} on {}, FDP {}".format(
+        args.workload, args.isa, "disabled" if args.disable_fdp else "enabled"
+    )
+)
+
+
+# The gem5 library simble board which can be used to run simple SE-mode
+# simulations.
+board = SimpleBoard(
+    clk_freq="3GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+# Here we set the workload. In this case we want to run a simple "Hello World!"
+# program compiled to the ARM ISA. The `Resource` class will automatically
+# download the binary from the gem5 Resources cloud bucket if it's not already
+# present.
+board.set_se_binary_workload(
+    # The `Resource` class reads the `resources.json` file from the gem5
+    # resources repository:
+    # https://gem5.googlesource.com/public/gem5-resource.
+    # Any resource specified in this file will be automatically retrieved.
+    # At the time of writing, this file is a WIP and does not contain all
+    # resources. Jira ticket: https://gem5.atlassian.net/browse/GEM5-1096
+    obtain_resource(workloads[args.workload][args.isa])
+)
+
+
+# Lastly we run the simulation.
+simulator = Simulator(board=board)
+simulator.run()
+
+print(
+    "Exiting @ tick {} because {}.".format(
+        simulator.get_current_tick(), simulator.get_last_exit_event_cause()
+    )
```

Comment:
```suggestion
    f"Exiting @ tick {simulator.get_current_tick()} because {simulator.get_last_exit_event_cause()}.".
```

Though, to be honest, this is a pretty useless print statement. I've been removing these from new scripts, personally.